### PR TITLE
refactor: move `ColladaExporter` into TypeScript

### PR DESCRIPTION
**Pull request by shuta13**
### What

Part of #12 

This is a PR of `ColladaExporter` migration to TypeScript.

### Checklist

- [ ] Ready to be merged

### CODE REVIEW

File: src/exporters/ColladaExporter.ts [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+export interface ColladaExporterResult {
+  data: string
+  textures: object[]
+}
+
+type GeometryInfo = { meshid: string; bufferGeometry: BufferGeometry }
+
+type MaterialPresentation = MeshPhongMaterial | MeshBasicMaterial | MeshLambertMaterial
+
+class ColladaExporter {
```
Comment (shuta13) [inline src/exporters/ColladaExporter.ts:45]: ref: https://github.com/mrdoob/three.js/blob/85490b18a0a171151d0f85500b051858c4634581/examples/jsm/exporters/ColladaExporter.js

**Review by shuta13 — COMMENTED** [review]
Please check!

File: src/exporters/ColladaExporter.ts [hunk 2]
```diff
@@ -0,0 +1,592 @@
...
+    // Convert to a uint8 array
+    return this.base64ToBuffer(base64data)
+  }
+
+  // gets the attribute array. Generate a new array if the attribute is interleaved
+  private attrBufferToArray(attr: InterleavedBufferAttribute | BufferAttribute): number[] | ArrayLike<number> {
+    if (attr instanceof InterleavedBufferAttribute && attr.isInterleavedBufferAttribute) {
+      // use the typed array constructor to save on memory
+      const TypedArrayConstructor: TypedArrayConstructors = attr.array.constructor
+      // @ts-expect-error
```
Comment (shuta13) [inline src/exporters/ColladaExporter.ts:234]: There are a few others, but I added `@ts-expect-error` as well as https://github.com/pmndrs/three-stdlib/blob/main/src/utils/BufferGeometryUtils.ts#L253

**Review by shuta13 — COMMENTED** [review]
There is one additional point.

**Comment by shuta13** [comment]
A few corrections have been made. Please review!

**Comment by shuta13** [comment]
A few corrections have been made. Please review!

**Review by joshuaellis — APPROVED** [review]

**Pull request merged**